File: design-pattern-jvm/app/src/main/java/design/pattern/java/patterns/behavioral/command/ServiceCommandBuilder.java
package design.pattern.java.patterns.behavioral.command;

public class ServiceCommandBuilder {
  public class ServiceCommand {
    private ServiceCommandBuilder command;

    private ServiceCommand(ServiceCommandBuilder command) {
      this.command = command;
    }

    public void Run() {
      this.command.service.apply(this.command.value);
    }
  }

  private String value;
  private Service service;

  public ServiceCommandBuilder(Service service) {
    this.service = service;
  }

  public ServiceCommand setArgs(String value) {
    this.value = value;
    return new ServiceCommand(this);
  }

}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/behavioral/command/ServiceCommandBuilder.kt
package design.pattern.kotlin.patterns.behavioral.command

class ServiceCommandBuilder(private val service: Service) {
    inner class ServiceCommand constructor(private val command: ServiceCommandBuilder) {
        fun Run() {
            command.service.apply(command.value)
        }
    }

    private var value: String? = null
    fun setArgs(value: String?): ServiceCommand {
        this.value = value
        return ServiceCommand(this)
    }
}